src/Utils/normalization.jl

```diff
@@ -1,68 +1,119 @@
+# linear ivp in canonical form x' = Ax
+const LCF = LCS{N, AN} where {N, AN<:AbstractMatrix{N}}
+
+# affine ivp with constraints in canonical form x' = Ax + u, u ∈ U, x ∈ X
+const ACF = CLCCS{N, AN, IdentityMultiple{N}, XT, UT} where {N, AN<:AbstractMatrix{N}, XT<:LazySet{N}, UT<:AbstractInput}
+
+# type union of canonical forms
+const CF = Union{<:LCF, <:ACF}
+
+# accepted types of non-deterministic inputs (non-canonical form)
+const UNCF = Union{<:LazySet{N}, Vector{<:LazySet{N}}, <:AbstractInput} where {N}
+
+# accepted types for the state constraints (non-canonical form)
+const XNCF = Union{<:LazySet{N}, Nothing} where {N}
+
 """
-    normalize(system)
+    normalize(system::AbstractContinuousSystem)
 
-Normalize a continuous system.
+Transform a continuous system to a normalized or canonical form.
 
 ### Input
 
 - `system` -- continuous system
 
 ### Output
 
-Either the same system if it already conforms to our required structure, or a
-new system otherwise.
+Either the same system if it already conforms to a canonical form, or a new
+system otherwise.
 
-### Algorithm
+### Notes
 
-We apply the following normalization steps.
+The normalization procedure consists of transforming a given system type into a
+"canonical" format that is used internally. The implementation of `normalize`
+exploits `MathematicalSystems`'s' types, which carry information about the problem
+as a type parameter.
 
-* [`normalize_inputs`](@ref)
-"""
-function normalize(system::InitialValueProblem{<:Union{AbstractContinuousSystem,
-                                                       AbstractDiscreteSystem}})
-    return normalize_inputs(system)
-end
+The type union `CF` defines the initial value problems considered canonical,
+i.e. which do not require normalization. More details are given below.
 
-"""
-    normalize_inputs(system)
+Homogeneous ODEs of the form ``x' = Ax`` are canonical if the associated
+problem is a `LinearContinuousSystem` and `A` is a matrix.
+Note that it does not consider constraints on the state-space (such as an
```

Comment:
```suggestion
Note that a `LinearContinuousSystem` does not consider constraints on the state-space (such as an
```

Reply:
Thanks. Actually there is the [ConstrainedLinearContinuousSystem](https://juliareach.github.io/MathematicalSystems.jl/latest/lib/types/#MathematicalSystems.ConstrainedLinearContinuousSystem), so we should support this as well. But i would rather program this in another PR.

Reply:
I just wanted to clarify the "it," that's all.